src/core/parser/PDFParser.ts

```diff
@@ -39,7 +39,9 @@ class PDFParser extends PDFObjectParser {
     this.objectsPerTick = objectsPerTick;
   }
 
-  async parseDocument(): Promise<PDFContext> {
+  async parseDocument(
+    throwOnInvalidObject: boolean = false,
```

Comment:
Passing this option through as a method parameter is pretty verbose, since it needs to be propagated through multiple methods before it is actually used. I'd prefer to have this be set as a `readonly` class property when the `PDFParser` instance is constructed. It would require fewer changes to be made and would be more consistent with the current API. 

Like this, for example:
```diff
diff --git a/src/core/parser/PDFParser.ts b/src/core/parser/PDFParser.ts
index d42713f..6b8b4ee 100644
--- a/src/core/parser/PDFParser.ts
+++ b/src/core/parser/PDFParser.ts
@@ -28,15 +28,22 @@ class PDFParser extends PDFObjectParser {
   static forBytesWithOptions = (
     pdfBytes: Uint8Array,
     objectsPerTick?: number,
-  ) => new PDFParser(pdfBytes, objectsPerTick);
+    throwOnInvalidObject?: boolean,
+  ) => new PDFParser(pdfBytes, objectsPerTick, throwOnInvalidObject);
 
   private readonly objectsPerTick: number;
+  private readonly throwOnInvalidObject: boolean;
   private alreadyParsed = false;
   private parsedObjects = 0;
 
-  constructor(pdfBytes: Uint8Array, objectsPerTick: number = Infinity) {
+  constructor(
+    pdfBytes: Uint8Array,
+    objectsPerTick = Infinity,
+    throwOnInvalidObject = false,
+  ) {
     super(ByteStream.of(pdfBytes), PDFContext.create());
     this.objectsPerTick = objectsPerTick;
+    this.throwOnInvalidObject = throwOnInvalidObject;
   }
 
   async parseDocument(): Promise<PDFContext> {
@@ -165,9 +172,9 @@ class PDFParser extends PDFObjectParser {
   private tryToParseInvalidIndirectObject() {
     const startPos = this.bytes.position();
 
-    console.warn(
-      `Trying to parse invalid object: ${JSON.stringify(startPos)})`,
-    );
+    const msg = `Trying to parse invalid object: ${JSON.stringify(startPos)})`;
+    if (this.throwOnInvalidObject) throw new Error(msg);
+    console.warn(msg);
 
     const ref = this.parseIndirectObjectHeader();

```

Reply:
Ah, it didn't event occur to me to store it in the readonly variables this does simplify the functions further down since they no longer need to explicitly passe on this value !